tools/querytee/proxy_test.go

```diff
@@ -159,25 +160,26 @@ func Test_Proxy_RequestsForwarding(t *testing.T) {
 
 			// Start the proxy.
 			cfg := ProxyConfig{
-				BackendEndpoints:   strings.Join(backendURLs, ","),
-				PreferredBackend:   strconv.Itoa(testData.preferredBackendIdx),
-				ServerServicePort:  0,
-				BackendReadTimeout: time.Second,
+				BackendEndpoints:      strings.Join(backendURLs, ","),
+				PreferredBackend:      strconv.Itoa(testData.preferredBackendIdx),
+				ServerServicePort:     80,
+				ServerGRPCServicePort: 90,
```

Comment:
It's a bit tricky setting hardcoded port numbers in the tests, because (a) you could have not privileges to open privileges ports (< 1024) and (b) whatever port you set, it could be already used causing tests to be flaky.

A trick frequently used in tests is to create a TCP socket to listen setting the port `0` so that the system auto-assign the port, then you close the socket, and you configure the test with that port (hoping in the meanwhile will not be taken by anyone else).

Reply:
Ah, that's interesting, so that means that using port `0` the system will automatically assign an open port?

Before the port was set to `0` and there was the `Endpoint()` method which would return the HTTP address, i.e., an address with the random port, e.g., `[::]:59966`. However, I was not able to find a method from weaveworks Server that would return the configured port 😕 I had to set the port here to guarantee that it is port 80. Do you know how I could get this information from the weaveworks server?

Reply:
Forget it 🤦‍♀️ already saw that there are `HTTPListenAddr()` and `GRPCListenAddr()` methods in weaveworks Server
https://github.com/weaveworks/common/blob/master/server/server.go#L503-L512